keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,295 @@
+Secret client library allows you to securely store and tightly control access to tokens, passwords, API keys, and other secrets. The library offers operations to create, retrieve, update, delete, purge, backup, restore and and list the secrets.
+
+Use the secret client library to create and manage secrets.
+
+[Source code][source_code] | [Package (Maven)][package] | [API reference documentation][api_documentation] | [Product documentation][azkeyvault_docs]
+
+## Getting started
+
+### Prerequisites
+
+- [Java Development Kit (JDK)][jdk] with version 8 or above
+- [Azure Subscription][azure_subscription]
+- An existing [Azure Key Vault][azure_keyvault] : Note that this document shows how to create one.
+
+### Adding the package to your project
+
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-keyvault-secrets</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+### Create an Azure Key Vault.
+
+To create an Azure Key Vault you can use the Azure Portal or [Azure Keyvault CLI][azure_keyvault_cli].
+
+Create the Azure Key Vault:
+```Powershell
+az keyvault create --name <keyvault-name> --resource-group <resource-group-name> --location eastus
+```
+
+### Authenticate the client
+
+Applications that use a key vault must authenticate by using a token from Azure Active Directory. The owner of the application must register it in Azure Active Directory first. At the end of registration, the application owner gets the following values:
+
+* An Application ID (also known as the AAD Client ID or appID)
+* An authentication key (also known as the shared secret).
+
+The application must present both these values to Azure Active Directory, to get a token. 
+
+#### Create/Get Credentials
+
+To create an application/service-principal you can use the [Azure Portal][azure_create_application_in_portal] or [Azure CLI][azure_cli].
+
+Create an application/service-principal in the Azure Active Directory.
+```Powershell
+az ad sp create-for-rbac -n <application-name> --password <application-password> --skip-assignment
+# If you don't specify a password, one will be created for you.
+```
+
+To authorize the same application to perform secret operations in your vault, type the following command:
+```Powershell
+az keyvault set-policy --name <keyvault-name> --spn <your-service-principal-id> --secret-permissions <secret-permissions>
+```
+
+#### Create Client
+
+In order to interact with the Azure Key Vault Secrets service you'll need to create an instance of the Secret Client class. To do this you'll need the application id and application key of an application in Azure Active Directory authorized with access to key vault.
+
+Once you have the values of the application id, application key and tenant id you can create the secret client.
+The following environment varaibles need to be configured to authorize with your key vault via default credentials.
```

Comment:
There need to be instructions on how to set up environment variables - definitely for bash and hopefully for windows cmd too. Described more in the comment above